packages/remix-server-runtime/entry.ts

```diff
@@ -7,15 +7,19 @@ import type {
 } from "./routes";
 import type { RouteData } from "./routeData";
 import type { RouteMatch } from "./routeMatching";
-import type { RouteModules, EntryRouteModule } from "./routeModules";
+import type {
+  RouteModules,
+  EntryRouteModule,
+  ServerRouteModule
+} from "./routeModules";
 
 export interface EntryContext {
   componentDidCatchEmulator: ComponentDidCatchEmulator;
   manifest: AssetsManifest;
   matches: RouteMatch<EntryRoute>[];
   routeData: RouteData;
   actionData?: RouteData;
-  routeModules: RouteModules<EntryRouteModule>;
+  routeModules: RouteModules<EntryRouteModule | ServerRouteModule>;
```

Comment:
Just realized that a `ServerRouteModule` extends `EntryRouteModule` so this isn't quite right. We probably want to have a type definition for a ResourceRouteModule or something 🤔